nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -234,4 +361,10 @@ public void process(final OutputStream out) throws IOException {
             }
         }
     }
+
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeChange(final PrimaryNodeState newState) {
+        isPrimaryNode = (newState == PrimaryNodeState.ELECTED_PRIMARY_NODE);
```

Comment:
This is a great catch, thank you!

@JPercivall in different PR I'm working on, #563 also needs similar functionality to know whether it's running on a standalone or clustered NiFi, and if latter whether it's a primary node.

I'm going to try adding a new `@OnAddedToCluster` lifecycle annotation taking one argument `isPrimaryNode`. Using it together with `@OnPrimaryNodeStateChange` Processors, Reporting tasks, and Controller services will be able to control its behavior well in a clustered environment.

How do you think?